examples/classification_example/main_train.py

```diff
@@ -0,0 +1,266 @@
+
+logger = logging.getLogger(__name__)
+
+logging.basicConfig(level=logging.INFO)
+
+device = torch.device("cuda") if torch.cuda.is_available() \
+    else torch.device("cpu")
+
+
+def construct_word_embedding_table(embed_dict, extractor: BaseExtractor):
+    embedding_dim = list(embed_dict.values())[0].shape[-1]
+
+    scale = np.sqrt(3.0 / embedding_dim)
+    table = np.empty(
+        [extractor.size(), embedding_dim], dtype=np.float32
+    )
+    oov = 0
+    for word, index in extractor.items():
+        if word in embed_dict:
+            embedding = embed_dict[word]
+        elif word.lower() in embed_dict:
+            embedding = embed_dict[word.lower()]
+        else:
+            embedding = np.random.uniform(
+                -scale, scale, [1, embedding_dim]
+            ).astype(np.float32)
+            oov += 1
+        table[index, :] = embedding
+    return torch.from_numpy(table)
+
+
+def create_model(text_extractor: AttributeExtractor,
+                 config: Config, in_channels: int):
+    embedding_dict = {}
+    fake_tensor = torch.tensor([0.0 for i in range(100)])
+    for word, index in text_extractor.items():
+        embedding_dict[word] = fake_tensor
+
+    word_embedding_table = \
+        construct_word_embedding_table(embedding_dict, text_extractor)
+
+    model: nn.Module = \
+        CNN_Classifier(in_channels=in_channels,
+                       word_embedding_table=word_embedding_table)
+
+    if config.config_model.model == "bert":
+        model: nn.Module = BERTClassifier()
+
+    return model, word_embedding_table
+
+
+def train(model: nn.Module, optim: Optimizer, batch: Batch, max_sen_length: int):
+    word = batch["text_tag"]["data"]
+    labels = batch["label_tag"]["data"]
+    optim.zero_grad()
+
+    logits, pred = None, None
+
+    if config.config_model.model == "cnn":
+        logits, pred = model(batch)
+
+    if config.config_model.model == "bert":
+        logits, pred = model(pad_each_bach(word, max_sen_length))
+
+    labels_1D = torch.squeeze(labels)
+
+    true_one_batch = (labels_1D == pred).sum().item()
+
+    loss = criterion(logits, labels_1D)
+
+    loss.backward()
+    optim.step()
+
+    batch_train_err = loss.item() * batch.batch_size
+
+    return batch_train_err, true_one_batch
+
+
+# All the configs
+config_data = yaml.safe_load(open("config_data.yml", "r"))
+config_model = yaml.safe_load(open("config_model.yml", "r"))
+
+config = Config({}, default_hparams=None)
+config.add_hparam('config_data', config_data)
+config.add_hparam('config_model', config_model)
+
+# Generate request
+text_extractor: AttributeExtractor = \
+    AttributeExtractor(config={"entry_type": Token,
+                               "vocab_method": "indexing",
+                               "attribute": "text"})
+
+
+class SentimentExtractor(AttributeExtractor):
+    def get_attribute(self, entry: Entry, attr: str):
+        if "positive" in getattr(entry, attr):
```

Comment:
```suggestion
   return "positive" if entry.sentiment["positive"] > entry.sentiment["negative"] else "negative"
```

First, you don't need to use `getattr` since this extractor is one time only. Second, you are not even consistent yourself, this function doesn't use `Dict`

Reply:
Maybe you misunderstand my logic here. Each data sample => a Sentence object => It has a Dictionary => It has only one key, negative or positive.

Reply:
No you are making an assumption that we shouldn't have made here. As we discussed, the format of the sentiment dict is that it can contain multiple keys. So let's simply don't assume only positive is there.

Reply:
Modified.